parkings/api/enforcement/active_parkings.py

```diff
@@ -0,0 +1,66 @@
+import datetime
+
+from django.db.models import Q
+from django.utils import timezone
+from rest_framework import generics, permissions, serializers
+
+from ...authentication import ApiKeyAuthentication
+from ...models import Parking
+from ..utils import parse_time
+
+
+class ActiveParkingsSerializer(serializers.ModelSerializer):
+    operator_name = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Parking
+        fields = [
+            'id',
+            'created_at',
+            'modified_at',
+            'registration_number',
+            'time_start',
+            'time_end',
+            'zone',
+            'operator_name',
+        ]
+
+    def get_operator_name(self, obj):
+        """
+        Get name of the operator of the parking.
+
+        :type obj: Parking
+        """
+        return obj.operator.name
+
+
+class ActiveParkingsView(generics.ListAPIView):
+    serializer_class = ActiveParkingsSerializer
+    authentication_classes = [ApiKeyAuthentication]
+    permission_classes = [permissions.IsAdminUser]
+    queryset = Parking.objects.order_by('-time_end')
+
+    def get_queryset(self):
+        qs = super(ActiveParkingsView, self).get_queryset()
+        reg_num = Parking.normalize_reg_num(
+            self.kwargs.get('registration_number'))
+        all_parkings_by_reg_num = qs.filter(normalized_reg_num=reg_num)
+        time_str = self.request.query_params.get('time')
+        time = parse_time(time_str) if time_str else timezone.now()
+        active_parkings = _filter_time_range(
+            all_parkings_by_reg_num,
+            starts_before=time,
+            ends_after=time)
+        if not active_parkings:
+            active_within_limit = _filter_time_range(
+                all_parkings_by_reg_num,
+                starts_before=time,
+                ends_after=(time - datetime.timedelta(days=1)))
+            return active_within_limit[:1]
+        return active_parkings
```

Comment:
It would be nicer if `FilterSet` from `django-filter` package was used instead of overriding `get_queryset()`. One obvious benefit would be consistent parsing of filtering values instead of needing to write your own `parse_time()`.

Reply:
I agree. I'll refactor it to use FilterSet.